pypeit/core/flux_calib.py

```diff
@@ -548,10 +549,81 @@ def sensfunc(wave, counts, counts_ivar, counts_mask, exptime, airmass, std_dict,
+    Args:
+        wave (float ndarray): shape = (nspec,)
+           Senstivity
+        wave_sens (float ndarray):
+           Sensfunc wavelength vector shape = (nsens,)
+        sensfunc (float ndarray): shape = (nsens,)
+           Sensitivity function
+        exptime (float):
+        telluric_correct:
+        extinct_correct:
+        airmass (float, optional):
+        longitude (float, optional):
+            longitude in degree for observatory
+        latitude:
+            latitude in degree for observatory
+            Used for extinction correction
+        extrap_sens (bool, optional):
+            Extrapolate the sensitivity function (instead of crashing out)
+
+    Returns:
+        sensfunc_interp/exptime/delta_wave: shape = (nspec,)
+
+    """
+
+    # TODO Telluric corrections via this method are deprecated
+    # Did the user request a telluric correction?
+    if telluric is not None:
+        # This assumes there is a separate telluric key in this dict.
+        msgs.info('Applying telluric correction')
+        sensfunc = sensfunc * (telluric > 1e-10) / (telluric + (telluric < 1e-10))
+
+    sensfunc_obs = np.zeros_like(wave)
+    wave_mask = wave > 1.0  # filter out masked regions or bad wavelengths
+    delta_wave = wvutils.get_delta_wave(wave, wave_mask)
+    try:
+        sensfunc_obs[wave_mask] = interpolate.interp1d(wave_sens, sensfunc, bounds_error=True)(wave[wave_mask])
+    except ValueError:
+        if extrap_sens:
+            sensfunc_obs[wave_mask] = interpolate.interp1d(wave_sens, sensfunc, bounds_error=False)(wave[wave_mask])
+            msgs.warn(
+                "Your data extends beyond the bounds of your sensfunc. You should be adjusting "
+                "the par['sensfunc']['extrap_blu'] and/or par['sensfunc']['extrap_red'] to extrapolate further "
+                "and recreate your sensfunc. But we are extrapolating per your direction. Good luck!")
+        else:
+            msgs.error("Your data extends beyond the bounds of your sensfunc. " + msgs.newline() +
+                       "Adjust the par['sensfunc']['extrap_blu'] and/or par['sensfunc']['extrap_red'] to extrapolate "
+                       "further and recreate your sensfunc.")
+
+    if extinct_correct:
+        if longitude is None or latitude is None:
+            msgs.error('You must specify longitude and latitude if we are extinction correcting')
+        # Apply Extinction if optical bands
+        msgs.info("Applying extinction correction")
+        msgs.warn("Extinction correction applyed only if the spectra covers <10000Ang.")
+        extinct = load_extinction_data(longitude, latitude)
+        ext_corr = extinction_correction(wave * units.AA, airmass, extinct)
+        senstot = sensfunc_obs * ext_corr
+    else:
+        senstot = sensfunc_obs.copy()
+
+    return senstot/exptime/delta_wave
+
+
 
 
 # JFH TODO This code needs to be cleaned up. The telluric option should probably be removed. Logic is not easy to follow.
```

Comment:
This is a good time to address this TODO.  Please do.

Reply:
I don't agree. More thought needs to be put into the path forward for these two different ways of flux calibrating.